woocommerce/woocommerce-gutenberg-products-block
# Unset default customer state if it doesn't match country

WooCommerce core has the option to set customer default country and state to store country and state. This seems to cause a problem in which if you select a different country, it will still try to get the store state and validate against it, this PR clear up the store state when that happens.



Fixes https://github.com/woocommerce/woocommerce-blocks/issues/8455




### Testing

1. Set US/California as the default shipping destination.
2. Add a product to the cart.
3. Go to the checkout page. 
4. Select a different Shipping Country that has states, e.g. Greece.
5. Fill in the city field.
6. You should not get a error.
7. Checkout without setting the state.
8. In the admin, you should not see California as the state.


### WooCommerce Visibility





### Changelog

> Fix cases in which Checkout would validate customer coutnry against the wrong state.
